blocks/block-edit/index.js

```diff
@@ -18,35 +19,47 @@ import {
 	hasBlockSupport,
 } from '../api';
 
-export function BlockEdit( props ) {
-	const { name, attributes = {} } = props;
-	const blockType = getBlockType( name );
-
-	if ( ! blockType ) {
-		return null;
+export class BlockEdit extends Component {
+	getChildContext() {
+		const blockType = getBlockType( this.props.name );
+		return { autocompleters: blockType.autocompleters };
```

Comment:
I see what you did. This is also a way to go.

Reply:
It seems like I missed something in our conversation yesterday, because I thought this is what we talked about. :)  Perhaps I missed your intent with the render-props pattern.

If we continue to use context, I'm thinking it would be cleaner and more generally useful to add the block type name or settings to the context rather than autocompleters specifically. Once you have the block type, it is straightforward to look up what you need.

Reply:
Context are fine, see you what I did today when testing React 16.3 and new Context API: https://github.com/WordPress/gutenberg/pull/5029.

I didn't mean we shouldn't do it. I just hoped we can move `<Autocomplete />` outside of `edit` component. It seems like it would be too tricky to achieve. Let's continue with the approach we have at the moment 👍

Reply:
Nice use of context IMO. I love the new API.